packages/next/client/image.tsx

```diff
@@ -49,18 +50,44 @@ type PlaceholderValue = 'blur' | 'empty'
 
 type ImgElementStyle = NonNullable<JSX.IntrinsicElements['img']['style']>
 
+interface ImageMetaData {
+  height: number
+  width: number
+}
+
+type StaticImageData = {
+  src: string
+} & ImageMetaData
+
+interface StaticRequire {
+  default: StaticImageData
+}
+
+type StaticImport = StaticRequire | StaticImageData
+
+function isStaticRequire(
+  data: StaticRequire | StaticImageData
+): data is StaticRequire {
+  return data && (data as StaticRequire).default !== undefined
+}
+
+function isUrl(data: string | StaticImport | URL): data is URL {
+  return !!data && (data as URL).href !== undefined
+}
+
 export type ImageProps = Omit<
   JSX.IntrinsicElements['img'],
   'src' | 'srcSet' | 'ref' | 'width' | 'height' | 'loading' | 'style'
 > & {
-  src: string
+  src: string | StaticImport | URL
   loader?: ImageLoader
   quality?: number | string
   priority?: boolean
   loading?: LoadingValue
   unoptimized?: boolean
   objectFit?: ImgElementStyle['objectFit']
   objectPosition?: ImgElementStyle['objectPosition']
+  meta?: ImageMetaData
```

Comment:
I don't see any tests for `meta`. How will this be used?

Reply:
Removed meta, since we're not supporting new URL() construction